plutus-playground-server/usecases/Game.hs

```diff
@@ -137,14 +137,15 @@ guess = endpoint @"guess" @GuessParams $ \(GuessParams theGuess) -> do
     void (submitTxConstraintsSpending gameInstance utxos tx)
 
 -- | Find the secret word in the Datum of the UTxOs
-findSecretWordValue :: Map TxOutRef ChainIndexTxOut -> Maybe HashedString
+findSecretWordValue :: Map TxOutRef OffChainTxOut -> Maybe HashedString
 findSecretWordValue =
   listToMaybe . catMaybes . Map.elems . Map.map secretWordValue
 
 -- | Extract the secret word in the Datum of a given transaction output is possible
-secretWordValue :: ChainIndexTxOut -> Maybe HashedString
+secretWordValue :: OffChainTxOut -> Maybe HashedString
 secretWordValue o = do
-  Datum d <- either (const Nothing) Just (_ciTxOutScriptDatum o)
+  -- FIXME is it important to check txOut is a script?
+  Datum d <- ocTxOutDatum o
```

Comment:
This funtion used `ciTxOutScriptDatum` which would only match when the txOut is protected by a script. Is this important? `ocTxOutDatum` on the other hand gives you a `Maybe Datum` independently from whether the txOut is protected by a pubkey or a script.

Reply:
Same here.